sources/mongodb/helpers.py

```diff
@@ -45,32 +46,66 @@ def __init__(
             self.last_value = None
 
     @property
-    def _filter_op(self) -> Dict[str, Any]:
+    def _sort_op(self) -> List[Optional[Tuple[str, int]]]:
         if not self.incremental or not self.last_value:
+            return []
+        if self.incremental.last_value_func is max:
+            return [(self.cursor_field, ASCENDING)]
+        elif self.incremental.last_value_func is min:
+            return [(self.cursor_field, DESCENDING)]
+        return []
+
+    @property
+    def _filter_op(self) -> Dict[str, Any]:
+        """Build a filtering operator.
+
+        Includes a field and the filtering condition for it.
+
+        Returns:
+            Dict[str, Any]: A dictionary with the filter operator.
+        """
+        if not (self.incremental and self.last_value):
             return {}
+
+        filt = {}
         if self.incremental.last_value_func is max:
-            return {self.cursor_field: {"$gte": self.last_value}}
+            filt = {self.cursor_field: {"$gte": self.last_value}}
+            if self.incremental.end_value:
+                filt[self.cursor_field]["$lt"] = self.incremental.end_value
+
         elif self.incremental.last_value_func is min:
-            return {self.cursor_field: {"$lt": self.last_value}}
-        return {}
+            filt = {self.cursor_field: {"$lte": self.last_value}}
+            if self.incremental.end_value:
+                filt[self.cursor_field]["$gt"] = self.incremental.end_value
 
-    def load_documents(self) -> Iterator[TDataItem]:
+        return filt
+
+    def load_documents(self, limit: Optional[int] = None) -> Iterator[TDataItem]:
+        """Construct the query and load the documents from the collection.
+
+        Args:
+            limit (Optional[int]): The number of documents to load.
+
+        Yields:
+            Iterator[TDataItem]: An iterator of the loaded documents.
+        """
         cursor = self.collection.find(self._filter_op)
+        if self._sort_op:
+            cursor = cursor.sort(self._sort_op)
+
+        if limit != 0:
```

Comment:
MongoDB allows negative limits. But if `0` is given, it'll use the default request size.